src/utils/handleArguments.js

```diff
@@ -0,0 +1,254 @@
+      return img.elt;
+    }
+  }
+  return null;
+}
+
+/**
+ * For methods which accept multiple optional arguments, a specific argument might be passed in multiple positions.
+ * We can determine which argument is which by examining their values.
+ *
+ * Creates an object where each argument is assigned to a named property.
+ * All properties are optional as arguments might be missing.
+ */
+
+/**
+ * @typedef {object} StandardArguments
+ * @property {string} [string] - A model name or any other string argument.
+ * @property {number} [number] - Any numeric argument.
+ * @property {function} [callback] - A callback function.
+ * @property {object} [options] - Any object which is not a media object is assumed to be options.
+ * @property {Array<any>} [array] - Any array.
+ * @property {HTMLMediaElement} [audio] - Both video and audio-only elements will be assigned to the audio property.
+ * @property {HTMLVideoElement} [video] - Video elements also get their own property.
+ * @property {InputImage} [image] - Any video, image, or image data.
+ */
+
+/**
+ * @class ArgHelper
+ * @implements {StandardArguments}
+ */
+class ArgHelper {
+  /**
+   * Arguments used to CREATE an image-based model can be:
+   *  - video: an HTMLVideoElement or p5 video.
+   *  - options: an object of options specific to this model.
+   *  - callback: a function to run once the model has been loaded. Called with arguments (error) or (error, result).
+   *  - modelName: some models accept a model name or URL as an argument.
+   *
+   * Arguments used to CALL a method an image-based model can be:
+   *  - image: an image or video element or an ImageData object.  Valid types: HTMLImageElement, HTMLCanvasElement,
+   *    HTMLVideoElement, ImageData, p5 image, p5 video.
+   *  - options: an object of options specific to this model.
+   *  - callback: a function to run once the method has been completed.
+   *
+   * Expected to be provided in order modelName, video/image, options, callback with any omitted.
+   * This function does not actually require any particular order.
+   *
+   * Later arguments will override earlier ones, so `this.video` should always be the first when providing arguments
+   * from a class method call.
+   *
+   *  @param {any[]} [args]
+   */
+  constructor(...args) {
+    args.forEach((arg) => this.addArg(arg));
+  }
+
+  /**
+   * Can add arguments through the constructor or at any time after construction.
+   *
+   * @param {any} arg
+   */
+  addArg(arg) {
+    // skip over falsey arguments and don't throw any error, assuming that these are omissions
+    // do this check first to prevent accessing properties on null, which is an object
+    if (arg === undefined || arg === null) {
+      return;
+    }
+    switch (typeof arg) {
+      case "string":
+        this.string = arg;
+        break;
+      case "number":
+        this.number = arg;
+        break;
+      case "function":
+        this.callback = arg
```

Comment:
Question:
* this might be something we defer to another time, but I'm wondering if you can imagine there every being a situation in which multiple functions are passed? and how we might handle such a case? I don't think this is something we'd see, but maybe one solution would be to say something like: "the last function passed in as an argument will be assigned to `this.callback`, if you, for whatever reason, need to pass in additional functions then add it to an object"

Reply:
There actually [is place in the `NeuralNetwork`](https://github.com/ml5js/ml5-library/blob/49f8d80861b0cd975dabc21ca9af345be31dea16/src/NeuralNetwork/index.js#L471) where we have multiple callbacks.

I don't know how we would handle that elegantly as it's hard to guess what the user intended.  It might be better to force the user to include additional functions as named properties on an options object so that we know that `options.whileTraining` is the `whileTraining` callback.

Due to the nature of "later arguments override earlier ones" your suggestion "the last function passed in as an argument will be assigned to this.callback" would be how it already works -- but the earlier functions would be discarded.  This is now making me wonder if we should write some sort of console warning in those cases.

Reply:
>  It might be better to force the user to include additional functions as named properties on an options object so that we know that options.whileTraining is the whileTraining callback.
+ This feels like a good way forward. I'm definitely all for making things as explicit as possible. 

>  This is now making me wonder if we should write some sort of console warning in those cases.
+ Oh yes! I think this would be a good compromise. Something like "Warning: it looks like you're trying to pass multiple functions to the X function. The X function only takes 1 callback. Any other functions should be a named property on the `options` object"

Reply:
I added a console warning for all duplicate properties of the same type. "`Received multiple ${property} arguments, but only a single ${property} is supported. The last ${property} will be used.`"